### ISSUE TITLE
Game Crashing when mod run on 1.20.1 Dedicated Server

### ISSUE THREAD

**Issue by Mortis10** [MarkusBordihn/BOs-Easy-NPC-Epic-Fight#9]
My game gives me "Internal Exception: io.netty.handler.codec.DecoderException:java.lang.ArrayIndexOutofBoundsException:Index 83 out of bounds for length 6" error and "crashes / kicks me from the server when I attempt to join it. I was able to play find until I was near the Epic Fight Easy NPC that I had attempted to place (which caused this error to initially pop up and refuse to stop). This error wasn't present on a Singleplayer world. 

**To Reproduce**

1. Host the mod on a dedicated server with proper versions
2. Place Epic Fight version of Easy NPC
3. Error (should) be encountered and prevent loading into the game / giving you the error

**Expected behavior**
Functional Epic Fight NPC

**Is the issue happening in combination with other mods? (If yes, please provide a full list)**
AzureLib 
GeckoLib 4 
YUNG's Bridges 
Placebo 
YUNG's Better Strongholds 
Temporary Spawners 
NetherPortalFix 
Simple Voice Chat 
Ender Dragon Fight Remastered 
Forgified Fabric API 
Just Enough Items 
Connector-1.0.0-beta.40+1.20.1 
Wood Makes Paper 
wthit 
YUNG's Better Desert Temples 
Balm 
brutalbosses mod 
Corpse 
WI Zoom 
Tree Harvester 
Bountiful 
Alternate Current 
Areas 
Epic Knights: Slavic Armory 
YUNG's API 
Bad Packets 
Aileron 
Dark Waters 
Library Of Exile 
Mouse Tweaks 
Terralith 
Highlighter 
Traveler's Titles 
YUNG's Menu Tweaks 
medievalmusic mod 
Cristel Lib 
Necronomicon 
XP Tome 
YUNG's Better Ocean Monuments 
Iceberg 
YUNG's Better Dungeons 
Searchables 
Towns and Towers 
EnhancedVisuals 
Potion Cauldron 
SmartBrainLib 
YUNG's Better Witch Huts 
GeckoLibIrisCompat 
YetAnotherConfigLib 
Camps. Castles. Carriages. 
Moonlight Library 
Collective 
Quality Sounds 
Epic Knights Mod 
YUNG's Better End Island 
Anonymous 
Citadel 
AI-Improvements 
Craftable Nametags 
Patchouli 
Small Ships 
Sound Physics Remastered 
Epic Knights: Addon 
Cloth Config v10 API 
Prism 
YDM's Weapon Master 
Continuity 
Sophisticated Core 
Workers Mod 
Shoulder Surfing 
Cupboard utilities 
Magnum Torch 
Yeetus Experimentus 
Sophisticated Backpacks 
Alex's Mobs 
Illager Invasion 
CreativeCore 
Beautified Chat Client 
YUNG's Better Jungle Temples 
kotlinforforge-4.10.0-all 
WF's Cave Overhaul 
Lootintegrations mod 
Kambrik 
Spawn Animations 
Medieval Buildings 
Controlling 
MidnightLib 
Dynamic Crosshair 
Nyf's Spiders 
Connectivity Mod 
Architectury 
AstikorCarts Redux 
Curios API 
YUNG's Better Nether Fortresses 
.connector 
Suggestion Provider Fix 
Simple Inventory Sorter 
WaveyCapes 
biomemusic mod 
YUNG's Extras 
Legendary Tooltips 
Recruits Mod 
Puzzles Lib 
Comforts 
YUNG's Better Mineshafts 
Presence Footsteps (Forge) 
Framework 
Extra Sounds 
Map Atlases 
Naturalist 
Mighty Mail 
Resourceful Lib 
AmbientSounds 
Dungeons and Taverns 
Handcrafted 
CraftTweaker 
Fantasy's Furniture 
Cuffed 
Epic Fight 
Easy NPC 
Bo's Easy NPC: Epic Fight 
Xaeros_Minimap_24.0.3_Forge_1.20 
notenoughanimations-forge-1.7.1-mc1.20.1 
textrues_embeddium_options-0.1.5+mc1.20.1 
radium-mc1.20.1-0.12.3+git.50c5c33 
betterfpsdist-1.20.1-4.3 
ImmediatelyFast-Forge-1.2.11+1.20.4 
embeddium-0.3.12+mc1.20.1 
acedium-0.2.6-beta 
dynamiclights-v1.7.1-mc1.17x-1.20x-mod 
CollisionFix-1.0.1 
XaerosWorldMap_1.38.1_Forge_1.20 
entity_model_features_forge_1.20.1-1.3 
modernfix-forge-5.15.0+mc1.20.1 
lazurite-1.0.4+mc1.20.1 
embeddiumplus-1.20.1-v1.2.5 
Log-Begone-Forge-1.20.1-1.0.8 
forgeskyboxes-0.0.2-1.20.2-new 
oculus-mc1.20.1-1.6.15a 
ferritecore-6.0.1-forge 
entity_texture_features_forge_1.20.1-5.2.3 
PacketFixer-forge-1.20.1-1.2.8
**Additional context**

I removed performance mods from my modlist and the error persisted.

**Comment by MarkusBordihn** [comment]
Thanks for the report. Could you please post the related crash log.

The error message basically means I received the 83 part of data, but only expected max. 6 parts in this case.
Which mean it could be related to any mod which is doing something with the network packets or or sending data.
However the crash log give more inside about the exact issue and the root cause.

**Comment by Mortis10** [comment]
No problem, I appreciate the quick response. I am not having full game crashes, neither server nor client side when using the (proper?) versions but the server log may be of some help? These are the logs related to Easy NPC

After viewing them myself I'll look into AristoCarts as it seems to be automatically using those NPCs, maybe causing the issues seeing as on the singleplayer world i hadn't placed one of those carts down, the only other honorable mention is with Villager Recruits I messed with Iron Golem spawn overrides (necessary to prevent despawning recruits)



[13Apr2024 08:29:01.352] [modloading-worker-0/INFO] [Easy NPC/]: Initializing Easy NPC (Forge) ...
[13Apr2024 08:29:01.352] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Debug Manager ...
[13Apr2024 08:29:01.353] [modloading-worker-0/WARN] [Easy NPC/]: [Debug Manager] Detected debug log level for Easy NPC with ALL!
[13Apr2024 08:29:01.353] [modloading-worker-0/WARN] [Easy NPC/]: [Debug Manager] Adjusting log level for Easy NPC from ALL to INFO, for performance reasons!
[13Apr2024 08:29:01.353] [modloading-worker-0/INFO] [Easy NPC/]: [Debug Manager] Add new logger config for Easy NPC with level INFO ...
[13Apr2024 08:29:01.354] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Constants ...
[13Apr2024 08:29:01.355] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Entities Types ...
[13Apr2024 08:29:01.363] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Blocks ...
[13Apr2024 08:29:01.364] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Blocks Entity Types ...
[13Apr2024 08:29:01.364] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Items ...
[13Apr2024 08:29:01.372] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Menu Handler ...
[13Apr2024 08:29:01.383] [modloading-worker-0/INFO] [Easy NPC/]: Registering Easy NPC common config ...
[13Apr2024 08:29:01.384] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Menu Types ...
[13Apr2024 08:29:01.387] [modloading-worker-0/INFO] [Easy NPC/]: 🗣 Register Easy NPC Network Message Handler ...
[13Apr2024 08:29:01.396] [modloading-worker-0/INFO] [Collective/]: [Collective] JSON file 'area_names.json' generation requested by mod 'areas'.
[13Apr2024 08:29:01.398] [modloading-worker-0/INFO] [Easy NPC/]: Registering Easy NPC client config ...


[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttackData
[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttackData
[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.344] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.AttributeData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.345] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.346] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ModelData
[13Apr2024 08:29:04.347] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.NavigationData
[13Apr2024 08:29:04.347] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ObjectiveData
[13Apr2024 08:29:04.347] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ObjectiveData
[13Apr2024 08:29:04.347] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ObjectiveData
[13Apr2024 08:29:04.347] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.OwnerData
[13Apr2024 08:29:04.347] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ProfessionData
[13Apr2024 08:29:04.347] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ScaleData
[13Apr2024 08:29:04.348] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ScaleData
[13Apr2024 08:29:04.348] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.ScaleData
[13Apr2024 08:29:04.348] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.SkinData
[13Apr2024 08:29:04.348] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.SkinData
[13Apr2024 08:29:04.348] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.SkinData
[13Apr2024 08:29:04.348] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.SkinData
[13Apr2024 08:29:04.348] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.SoundData
[13Apr2024 08:29:04.349] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.TradingData
[13Apr2024 08:29:04.349] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.TradingData
[13Apr2024 08:29:04.349] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.TradingData
[13Apr2024 08:29:04.349] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.TradingData
[13Apr2024 08:29:04.349] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.TradingData
[13Apr2024 08:29:04.349] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.TradingData
[13Apr2024 08:29:04.349] [main/WARN] [net.minecraft.network.syncher.SynchedEntityData/]: defineId called for: class de.markusbordihn.easynpc.entity.EasyNPCBaseEntity from interface de.markusbordihn.easynpc.entity.easynpc.data.VariantData
[13Apr2024 08:29:04.626] [main/WARN] [net.minecraftforge.common.ForgeConfigSpec/CORE]: Configuration file /home/container/config/easy_npc/easy_npc-common.toml is not correct. Correcting

pull_animals = [
    "minecraft:camel",
    "minecraft:donkey",
    "minecraft:horse",
    "minecraft:mule",
    "minecraft:skeleton_horse",
    "minecraft:zombie_horse",
    "minecraft:player",
    "easy_npc:humanoid",
    "easy_npc:humanoid_slim",
    "easy_npc:allay",
    "easy_npc:cat",
    "easy_npc:chicken",
    "easy_npc:fairy",
    "easy_npc:iron_golem",
    "easy_npc:skeleton",
    "easy_npc:stray",
    "easy_npc:wither_skeleton",
    "easy_npc:villager",
    "easy_npc:zombie_villager",
    "easy_npc:husk",
    "easy_npc:drowned",
    "easy_npc:zombie",
    "easy_npc:pig",
    "easy_npc:evoker",
    "easy_npc:illusioner",
    "easy_npc:pillager",
    "easy_npc:vindicator",
    "easy_npc:orc",
    "easy_npc:orc_warrior",
    "easy_npc:wolf",
    "easy_npc:horse",
    "easy_npc:skeleton_horse",
    "easy_npc:zombie_horse",
    "naturalist:zebra",
    "easy_npc_epic_fight:evoker",
    "easy_npc_epic_fight:humanoid",
    "easy_npc_epic_fight:humanoid_slim",
    "easy_npc_epic_fight:husk",
    "easy_npc_epic_fight:illusioner",
    "easy_npc_epic_fight:iron_golem",
    "easy_npc_epic_fight:skeleton",
    "easy_npc_epic_fight:stray",
    "easy_npc_epic_fight:pillager",
    "easy_npc_epic_fight:vindicator",
    "easy_npc_epic_fight:wither_skeleton",
    "easy_npc_epic_fight:zombie",
    "easy_npc_epic_fight:zombie_villager"
  ]


[13Apr2024 08:29:07.908] [main/WARN] [epicfight/]: [Custom Entity] Entity named easy_npc_epic_fight:iron_golem_ does not exist

[13Apr2024 08:30:13.482] [Server thread/INFO] [Easy NPC/]: Define synced data for ZombieEF['entity.easy_npc_epic_fight.zombie'/523, l='ServerLevel[world]', x=0.00, y=0.00, z=0.00]
[13Apr2024 08:30:13.483] [Server thread/INFO] [Easy NPC/]: Register default handler for ZombieEF['entity.easy_npc_epic_fight.zombie'/523, l='ServerLevel[world]', x=0.00, y=0.00, z=0.00] with variant ZOMBIE ...
[13Apr2024 08:30:13.483] [Server thread/INFO] [Easy NPC/]: Define custom server-side data for ZombieEF['entity.easy_npc_epic_fight.zombie'/523, l='ServerLevel[world]', x=0.00, y=0.00, z=0.00] ...
[13Apr2024 08:30:13.489] [Server thread/INFO] [Easy NPC/]: Spawned Easy NPC Entity ZombieEF['entity.easy_npc_epic_fight.zombie'/523, l='ServerLevel[world]', x=7.50, y=84.00, z=1556.50] by player ServerPlayer['Sablemane'/251, l='ServerLevel[world]', x=7.20, y=84.00, z=1558.03] ...
[13Apr2024 08:30:13.531] [Server thread/INFO] [net.minecraft.server.network.ServerGamePacketListenerImpl/]: Sablemane lost connection: Disconnected
[13Apr2024 08:30:13.532] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: Sablemane left the game
[13Apr2024 08:30:13.533] [Server thread/INFO] [voicechat/]: [voicechat] Disconnecting client Sablemane
[13Apr2024 08:30:13.637] [Server thread/INFO] [Easy NPC/]: Add additional save data for ZombieEF['entity.easy_npc_epic_fight.zombie'/523, l='ServerLevel[world]', x=7.50, y=84.00, z=1556.50]

**Comment by Mortis10** [comment]
NOTE: Tried altering the carts mod, had no effect on the issue.

**Comment by Mortis10** [comment]
I'm by no means knowledgeable on anything code related but perhaps those Data calls are what the issue is? Also, is the latest Epic Fight version compatible? And if not which one should I be using for 1.20.1 with the latest of Easy NPC / Easy NPC EF

**Comment by MarkusBordihn** [comment]
Thanks for the additional details.
The server logs shows nothing strange but confirms that this is a client side related issue and not a server side one.

I tried to reproduce the issue myself and was not able to reproduce the issue.
![image](https://github.com/MarkusBordihn/BOs-Easy-NPC-Epic-Fight/assets/12183293/181c716d-f820-459b-89b4-869081c31f9a)

Did you used the spawn NPC eggs or did you used the `/summon` command ?
Some of the NPCs are WIP and only available over the `/summon` command and so could cause crashes like the Iron Golem.

Currently I assume it's related to one of the network optimizer mods, which try to fix something but failed to do.

I used the following versions for testing:
- https://www.curseforge.com/minecraft/mc-mods/easy-npc/files/5247170
- https://www.curseforge.com/minecraft/mc-mods/epic-fight-mod/files/5159623

It's strange that you don't get any crash log at all. 
You can try to enable the debug logs over the CurseForge launcher or so, to see if they give any additional hints.

Unfortunately at the moment I'm unable to reproduce or confirm the issue from my side. :/

**Comment by Mortis10** [comment]
I was using the NPC spawn eggs, namely Humanoid and Zombie. Which version of Easy NPC are you using for those other two mods? I'm currently using the latest one, is that also compatible?
Thank you for your assistance

**Comment by Mortis10** [comment]
Also the Easy NPC Epic Fight compat version, please

**Comment by Mortis10** [comment]
Sorry, sleep deprivation lol. I have the mod versions that you have linked already, and using the 4.0 version of the Compatibility mod. I'll see if I can get more logs and try /summon or removing more mods. Ill update you ASAP

**Comment by Mortis10** [comment]
[06:46:25] [main/ERROR] [ne.mi.fm.lo.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/model/HumanoidModel for invalid dist DEDICATED_SERVER


Any idea if that might have to do with it?

**Comment by Mortis10** [comment]
I managed to get it to work! If you'd like I can try to see specifically which mod was incompatible, but when I disabled Sinyatra Connector (and a few other Fabric mods being run, listed : Lazurite, Continuity, Forgified Fabric API, Collision Fix <--- Probable Culprit) it now lets me load in just fine. So i'm assuming there's some sort of major incompatibility with one of if not all of those mods, which isn't a big deal I'm just glad I figured it out lol

Once again, thank you for your help and testing it out.
